smdebug/core/hook.py

```diff
@@ -660,16 +663,32 @@ def export_collections(self):
         collection_file_name = f"{self.worker}_collections.json"
         self.collection_manager.export(self.out_dir, collection_file_name)
 
-    def _get_reduction_tensor_name(self, tensor_name, reduction_name, abs):
-        return get_reduction_tensor_name(tensor_name, reduction_name, abs, remove_colon_index=True)
+    def _get_reduction_tensor_name(self, tensor_name, reduction_name, abs, collection_name=""):
+        return get_reduction_tensor_name(
+            tensor_name,
+            reduction_name,
+            abs,
+            remove_colon_index=True,
+            collection_name=collection_name,
+        )
 
-    def _write_reduction(self, tensor_name, tensor_value, reduction_name, abs, tensor_ref=None):
-        reduction_tensor_name = self._get_reduction_tensor_name(tensor_name, reduction_name, abs)
+    def _write_reduction(
+        self, tensor_name, tensor_value, reduction_name, abs, tensor_ref=None, collection_name=""
+    ):
+        reduction_tensor_name = self._get_reduction_tensor_name(
+            tensor_name, reduction_name, abs, collection_name=collection_name
+        )
         try:
             tensor_data = self._get_reduction_of_data(
                 reduction_name, tensor_value, tensor_name, abs
             )
             self._write_raw_tensor_simple(reduction_tensor_name, tensor_data, tensor_ref=tensor_ref)
+            if abs:
+                reduction_name = "abs_" + reduction_name
+            tb_writer = self._maybe_get_tb_writer(subfolder=reduction_name)
+            if tb_writer:
+                scalar = self._make_numpy_array(tensor_data)
```

Comment:
Per default Debugger writes reductions like normal tensors into debug-output folder and users can retrieve the data via the smdebug API. I extended this part, so that reductions are also written in Tensorboard format (in case user provided a Tensorboard configuration)